bindings/python/pyiree/rt/system_api.py

```diff
@@ -125,6 +133,41 @@ def _get_global_config():
   return _global_config
 
 
+def _normalize_numpy(array: np.ndarray,
+                     single_precision: bool = False) -> np.ndarray:
+  if np.isscalar(array):
+    array = np.array(array)
+
+  # IREE models booleans as int8s.
+  if array.dtype == np.bool:
+    array = array.astype(np.int8)
+
+  if single_precision:
+    if array.dtype == np.float16 or array.dtype == np.float64:
+      array = array.astype(np.float32)
+    if array.dtype == np.int16 or array.dtype == np.int64:
+      array = array.astype(np.int32)
+  return array
+
+
+def normalize_value(value: Any) -> Optional[np.ndarray]:
+  """Normalizes the given value for input to (or comparison with) IREE."""
+  if isinstance(value, np.ndarray):
+    return _normalize_numpy(value)
+  elif isinstance(value, (bool, int, float, list, tuple)):
+    # Always use 32-bits to represent Python inputs.
+    return _normalize_numpy(np.array(value), single_precision=True)
+  elif hasattr(value, "numpy"):
+    # Converts TensorFlow and Torch tensors.
+    return _normalize_numpy(value.numpy())
+  elif value is None:
```

Comment:
Nit: do the is None check first as it is (far) lighter weight than the isinstance and hasattr checks.